userspace/libscap/scap.c

```diff
@@ -968,6 +968,8 @@ scap_t* scap_open_plugin_int(char *error, int32_t *rc, source_plugin_info* input
 	memset(handle, 0, sizeof(scap_t));
 	handle->m_mode = SCAP_MODE_PLUGIN;
 
+	handle->plugin_api_vers = PPM_PLUGIN_VERSION(PLUGIN_API_VERSION_MAJOR, PLUGIN_API_VERSION_MINOR, PLUGIN_API_VERSION_PATCH);
```

Comment:
It's kind of a wierd split between the plugin object up in libsinsp and the scap code that actually opens a stream of events from a plugin, but I think it would be better to limit the plugin code in libscap as much as possble.

Currently, all the work of opening the shared library, resolving symbols, version checking, etc happens in libsinsp, via sinsp_plugin::register_plugin/sinsp::add_plugin. The only thing passed down to libscap is the struct with resolved symbols in sinsp::open_live_common.

Given that, to me it makes the most sense to encode the framework api version as a static sinsp_plugin::version object rather than having a scap call to fetch it, when the thing coming back will be compared to a sinsp_plugin::version object anyway.

Does this seem reasonable?

Reply:
> Given that, to me it makes the most sense to encode the framework api version as a static sinsp_plugin::version object rather than having a scap call to fetch it, when the thing coming back will be compared to a sinsp_plugin::version object anyway.

I agree, in the end it seems the correct way to go.

Reply:
@mstemm  the changes are now in place ;) Thanks again!